### DISCUSSION TITLE
let tabopen also shows opened tabs

### DISCUSSION THREAD

**Discussion by NestorLiao** [tridactyl/tridactyl#5071] [Ideas]
It is convenient to have both tab and tabopen function works together, because people sometimes forget wether the site is opened before. In vimium, you can do something like tabopen, but also arrange the currently opend tab on top. if regex get opened site, then jump to it, if you current tabs don't have one, it will open that site by history regex or search engine.  Then we don't need to b->esc->t anymore.

**Comment by bovine3dom** [comment]
Yeah, I've wanted to do this for years, but it would be tricky to do with the current architecture. I mentioned it briefly here - https://github.com/tridactyl/tridactyl/issues/4709 ("multiple completions should display at once")

For your specific usecase you might get by with something like

```
command mytabopen js -d` tri.excmds.tab(JS_ARGS.slice(1)).catch(e => tri.excmds.fillcmdline("tabopen", ...JS_ARGS))`
```

which when you press enter will try to go to that tab, otherwise it will open up `:tabopen`. It's a bit janky and you could probably improve it :)
